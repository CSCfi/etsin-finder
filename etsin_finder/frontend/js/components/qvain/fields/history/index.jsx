import React from 'react'
import { Section } from '../../general/section/index'
import TooltipContent from './TooltipContent'
import RelatedResource from './relatedResource'
import Infrastructure from './infrastructure'
import Provenance from './provenance'

const translations = {
  title: 'qvain.history.title',
  tooltip: 'qvain.history.tooltip',
}

const components = {
  tooltipContent: TooltipContent,
}

const fieldProps = {
  translations,
  components,
}

const History = () => (
  <Section {...fieldProps}>
    <RelatedResource />
    <Infrastructure />
    <Provenance />
  </Section>
)

export default History
