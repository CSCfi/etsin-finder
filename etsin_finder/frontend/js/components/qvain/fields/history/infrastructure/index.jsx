import React from 'react'
import InfrastructureSelection from './InfrastructureSelection'
import { Field } from '../../../general/section'

const brief = {
  title: 'qvain.history.infrastructure.title',
  description: 'qvain.history.infrastructure.description',
}

export default () => (
  <Field brief={brief} labelFor="infrastructure-select">
    <InfrastructureSelection />
  </Field>
)
