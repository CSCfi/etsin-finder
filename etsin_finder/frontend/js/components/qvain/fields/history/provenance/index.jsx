import React from 'react'
import PropTypes from 'prop-types'
import { inject, observer } from 'mobx-react'
import { Field } from '../../../general/section'
import ProvenanceFieldContent from './ProvenanceFieldContent'

const brief = {
  title: 'qvain.history.provenance.title',
  description: 'qvain.history.provenance.description',
}

const Provenance = ({ Stores }) => {
  const Store = Stores.Qvain
  const { lang } = Stores.Locale
  return (
    <Field brief={brief}>
      <ProvenanceFieldContent Store={Store} lang={lang} />
    </Field>
  )
}

Provenance.propTypes = {
  Stores: PropTypes.object.isRequired,
}

export default inject('Stores')(observer(Provenance))
