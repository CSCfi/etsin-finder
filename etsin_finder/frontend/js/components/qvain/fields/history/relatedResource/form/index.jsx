import React, { useState } from 'react'
import PropTypes from 'prop-types'
import TranslationTab from '../../../../general/input/translationTab'
import TabInput from '../../../../general/input/translationTabInputModal'
import { FormContainer } from '../../../../general/modal/form'
import ModalInput from '../../../../general/modal/modalInput'
import ModalReferenceInput from '../../../../general/modal/modalReferenceInput'
import { RelationType } from '../../../../../../stores/view/qvain/qvain.relatedResources'
import ModalSeparator from '../../../../general/modal/modalSeparator'
import { useStores } from '../../../../utils/stores'

const Form = props => {
  const {
    Locale: { getMatchingLang },
  } = useStores()
  const { inEdit } = props.Field

  const translations = [inEdit.name, inEdit.description]
  const [language, setLanguage] = useState(getMatchingLang(translations))

  return (
    <FormContainer>
      <TranslationTab language={language} setLanguage={setLanguage}>
        <TabInput {...props} datum="name" language={language} isRequired />
        <TabInput {...props} datum="description" language={language} />
      </TranslationTab>
      <ModalInput {...props} datum="identifier" />
      <ModalReferenceInput
        {...props}
        datum="entityType"
        metaxIdentifier="resource_type"
        model={RelationType}
      />
      <ModalSeparator />
      {!props.hideRelationType && (
        <ModalReferenceInput
          {...props}
          datum="relationType"
          metaxIdentifier="relation_type"
          model={RelationType}
          isRequired
        />
      )}
    </FormContainer>
  )
}

Form.propTypes = {
  Field: PropTypes.object.isRequired,
  translationsRoot: PropTypes.string.isRequired,
  hideRelationType: PropTypes.bool,
}

Form.defaultProps = {
  hideRelationType: false,
}

export default Form
