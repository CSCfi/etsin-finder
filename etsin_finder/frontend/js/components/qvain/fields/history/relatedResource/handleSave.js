import { relatedResourceNameSchema, relatedResourceTypeSchema } from '../../../utils/formValidation'

export default async (Field, options = {}) => {
  const { inEdit, save, clearInEdit, setValidationError } = Field

  try {
    await relatedResourceNameSchema.validate(inEdit.name)
    if (!options.noRelationType) {
      await relatedResourceTypeSchema.validate(inEdit.relationType)
    }

    if ((inEdit.name.fi || inEdit.name.en) && !inEdit.name.und) {
      inEdit.name.und = inEdit.name.fi || inEdit.name.en
    }

    await save()
    clearInEdit()
  } catch (e) {
    setValidationError(e.message)
  }
}
