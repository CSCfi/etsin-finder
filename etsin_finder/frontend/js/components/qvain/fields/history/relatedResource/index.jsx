import React from 'react'

import { withFieldErrorBoundary } from '../../../general/errors/fieldErrorBoundary'
import { Field } from '../../../general/section'
import RelatedResourceContent from './relatedResourceContent'

const brief = {
  title: 'qvain.history.relatedResource.title',
  description: 'qvain.history.relatedResource.description',
}

export const RelatedResource = () => (
  <Field brief={brief}>
    <RelatedResourceContent />
  </Field>
)

export default withFieldErrorBoundary(RelatedResource, brief.title)
