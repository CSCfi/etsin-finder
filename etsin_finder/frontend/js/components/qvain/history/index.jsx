import React, { Component } from 'react'
import Field from '../general/field'
import TooltipContent from './TooltipContent'
import Infrastructure from './infrastructure'

const translations = {
  title: 'qvain.history.title',
  tooltip: 'qvain.history.tooltip',
}

const components = {
  TooltipContent,
}

const fieldProps = {
  translations,
  components,
}

class History extends Component {
  render() {
    return (
      <Field {...fieldProps}>
        <Infrastructure />
      </Field>
    )
  }
}

export default History
