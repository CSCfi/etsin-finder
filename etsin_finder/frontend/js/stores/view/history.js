import { observable, computed, action } from 'mobx'
import createHistory from 'history/createBrowserHistory'

const history = createHistory()

class History {
  @observable history = history

  @computed
  get urlParams() {
    const params = {}
    const search = this.history.location.search.slice(1)
    const definitions = search.split('&')
    definitions.forEach(val => {
      const parts = val.split('=', 2)
      params[parts[0]] = parts[1]
    })
    return search !== '' ? params : null
  }

  @action
  setUrlParams = params => {
    let url = '?'
    const keys = Object.keys(params)
    for (let i = 0; i < keys.length; i += 1) {
      console.log(params[keys[i]])
      url = `${url}${keys[i]}=${params[keys[i]]}`
      if (i + 1 < keys.length) url = `${url}&` // not on last
    }
    this.history.push({
      pathname: this.history.location.pathname,
      search: url,
    })
  }
}

export default new History()
