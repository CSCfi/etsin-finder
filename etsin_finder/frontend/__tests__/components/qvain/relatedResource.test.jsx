import React from 'react'
import { shallow } from 'enzyme'
import { RelatedResource } from '../../../js/components/qvain/fields/history/relatedResource'

describe('RelatedResource', () => {
  let wrapper
  let Field
  beforeEach(() => {
    wrapper = shallow(<RelatedResource />)
    Field = wrapper.find('Field')
  })
  test('it renders', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('it renders Field', () => {
    expect(Field.debug()).toBeTruthy()
  })
})
