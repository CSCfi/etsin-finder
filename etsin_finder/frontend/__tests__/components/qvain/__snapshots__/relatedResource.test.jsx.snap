// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedResource it renders 1`] = `
<Field
  brief={
    Object {
      "description": "qvain.history.relatedResource.description",
      "title": "qvain.history.relatedResource.title",
    }
  }
>
  <RelatedResourceContent />
</Field>
`;
